src/app/contexts/EventTrackingContext/index.test.jsx

```diff
@@ -0,0 +1,171 @@
+    render(
+      <Wrapper>
+        <TestComponent />
+      </Wrapper>,
+    );
+
+    const testEl = screen.getByTestId('test-component');
+    const trackingData = JSON.parse(testEl.textContent);
+
+    expect(trackingData).toEqual({
+      campaignName: 'article-sty',
+      pageIdentifier: 'news::pidgin.news.story.51745682.page',
+      platform: 'canonical',
+      statsDestination: 'WS_NEWS_LANGUAGES_TEST',
+    });
+  });
+
+  it('should provide an empty object if the eventTracking toggle is disabled', () => {
+    const eventTrackingToggle = {
+      eventTracking: {
+        enabled: false,
+      },
+    };
+
+    render(
+      <Wrapper toggles={eventTrackingToggle}>
+        <TestComponent />
+      </Wrapper>,
+    );
+
+    const testEl = screen.getByTestId('test-component');
+    const trackingData = JSON.parse(testEl.textContent);
+
+    expect(trackingData).toEqual({});
+  });
+});
+
+describe('Error handling', () => {
+  it('should not throw error and not log error when no pageData is passed into context provider', async () => {
+    let errorMessage;
+
+    try {
+      render(
+        <Wrapper pageData={null}>
+          <TestComponent />
+        </Wrapper>,
+      );
+    } catch ({ message }) {
+      errorMessage = message;
+    }
+
+    const testEl = screen.getByTestId('test-component');
+    const trackingData = JSON.parse(testEl.textContent);
+
+    expect(trackingData).toEqual({});
+    expect(errorMessage).toBeUndefined();
+    expect(global.fetch).not.toHaveBeenCalled();
+  });
+
+  it('should not throw error when unexpected data is passed into context provider', async () => {
+    let errorMessage;
+
+    try {
+      render(
+        <Wrapper pageData={['unexpected data']}>
+          <TestComponent />
+        </Wrapper>,
+      );
+    } catch ({ message }) {
+      errorMessage = message;
+    }
+
+    const testEl = screen.getByTestId('test-component');
+    const trackingData = JSON.parse(testEl.textContent);
+
+    expect(trackingData).toEqual({});
+    expect(errorMessage).toBeUndefined();
+    expect(console.error).toHaveBeenCalledWith(
+      expect.stringContaining(
+        'ATI Event Tracking Error: Could not parse tracking values from page data:',
+      ),
+    );
+    expect(global.fetch).not.toHaveBeenCalled();
+  });
+
+  it('should not provide tracking props when there is no page type campaign ID', async () => {
+    let errorMessage;
+
+    try {
+      render(
+        <Wrapper pageType="funky-page-type">
+          <TestComponent />
+        </Wrapper>,
+      );
+    } catch ({ message }) {
+      errorMessage = message;
+    }
+
+    const testEl = screen.getByTestId('test-component');
+    const trackingData = JSON.parse(testEl.textContent);
+
+    expect(trackingData).toEqual({});
+    expect(errorMessage).toBeUndefined();
+    expect(console.error).toHaveBeenCalledWith(
+      expect.stringContaining(
+        "ATI Event Tracking Error: Could not get the page type's campaign name",
```

Comment:
Further to your comment above, can see campaign name and campaign id being used interchangibly, may be good to pick one for readability?